### PR TITLE
fix: upgrade web3 dependency to v4 in defender sdk

### PR DESCRIPTION
**Pull request by MCarlomagno**
# Summary
Upgraded web3 js version from v1 to v4. 

Adapted provider API to support [EIP-1193](https://eips.ethereum.org/EIPS/eip-1193)

## Testing Process
- [ ] run relayer-web3-provider example
- [ ] run unit tests

### CODE REVIEW

**Review by tirumerla — APPROVED** [review]
Lgtm, thanks. Once we release this sdk version, we can update the backend version for action layer

File: packages/relay-signer/src/web3/query.ts [hunk 1]
```diff
@@ -1,31 +1,22 @@
...
-      .call({ method: payload.method, params: payload.params ?? [] })
-      .then((response) =>
-        callback(null, {
-          ...response,
-          id: payloadId ?? response.id ?? this.id++,
-        }),
-      )
-      .catch((err) => callback(err, undefined));
+  public async request<T>(payload: JsonRpcRequest<string[]>): Promise<JsonRpcResponseWithResult<T>> {
+    const toJsonRpcResponse = (result: any): JsonRpcResponseWithResult<T> => ({
```
Comment (zeljkoX) [inline packages/relay-signer/src/web3/query.ts:15]: this could be utility fn and not nested one

File: packages/relay-signer/src/web3/sender.ts [hunk 2]
```diff
@@ -30,17 +35,18 @@ export type DefenderRelaySenderOptions = Partial<{
   maxPriorityFeePerGas: BigUInt;
   speed: Speed;
   validForSeconds: number;
+  authConfig: AuthConfig;
```
Comment (zeljkoX) [inline packages/relay-signer/src/web3/sender.ts:38]: do we need authConfig here? used for logic from web3 module?

**Review by zeljkoX — APPROVED** [review]
LGTM

Thanks!

File: packages/relay-signer/src/web3/sender.ts (hunk 2 above)
Comment (MCarlomagno) [reply packages/relay-signer/src/web3/sender.ts:38]: yup, it's a leftover

File: packages/relay-signer/src/web3/query.ts (hunk 1 above)
Comment (MCarlomagno) [reply packages/relay-signer/src/web3/query.ts:15]: I just realized we don't even need that function, it was just a trick to fix the linter. Using `as Promise<JsonRpcResponseWithResult<T>>` works

**Pull request merged**